### PR TITLE
[mlir][VectorToSPIRV] Add conversion for vector.extract with dynamic indices

### PR DESCRIPTION
**Pull request by Groverkss**

### CODE REVIEW

File: mlir/lib/Conversion/VectorToSPIRV/VectorToSPIRV.cpp [hunk 1]
```diff
@@ -249,9 +242,20 @@ struct VectorInsertOpConvert final
...
 
-    int32_t id = getFirstIntValue(insertOp.getMixedPosition());
-    rewriter.replaceOpWithNewOp<spirv::CompositeInsertOp>(
-        insertOp, adaptor.getSource(), adaptor.getDest(), id);
+    std::optional<int64_t> id =
+        getConstantIntValue(insertOp.getMixedPosition()[0]);
+
+    //    rewriter.replaceOpWithNewOp<spirv::CompositeInsertOp>(
+    //        insertOp, adaptor.getSource(), adaptor.getDest(), id);
+    //    return success();
```
Comment (kuhar) [inline mlir/lib/Conversion/VectorToSPIRV/VectorToSPIRV.cpp:250]: Delete this

File: mlir/test/Conversion/VectorToSPIRV/vector-to-spirv.mlir [hunk 2]
```diff
@@ -216,6 +236,28 @@ func.func @insert_size1_vector(%arg0 : vector<1xf32>, %arg1: f32) -> vector<1xf3
...
+}
+
+// -----
+
+// CHECK-LABEL: @insert_dynamic_cst
+//  CHECK-SAME: %[[VAL:.*]]: f32, %[[V:.*]]: vector<4xf32>
+//       CHECK:   spirv.CompositeInsert %[[VAL]], %[[V]][2 : i32] : f32 into vector<4xf32>
+func.func @insert_dynamic_cst(%val: f32, %arg0 : vector<4xf32>) -> vector<4xf32> {
+  %idx = arith.constant 2 : index
+  %0 = vector.insert %val, %arg0[%idx] : f32 into vector<4xf32>
```
Comment (kuhar) [inline mlir/test/Conversion/VectorToSPIRV/vector-to-spirv.mlir:255]: Can you add a lit test for the dynamic scalar case? I don't see it in this file.

Same for extract.

**Review by kuhar — CHANGES_REQUESTED** [review]

File: mlir/test/Conversion/VectorToSPIRV/vector-to-spirv.mlir (hunk 2 above)
Comment (kuhar) [reply mlir/test/Conversion/VectorToSPIRV/vector-to-spirv.mlir:255]: By 'the scalar case' I mean `vector<1xT>` which becomes `T` in SPIR-V

Comment (Groverkss) [reply mlir/test/Conversion/VectorToSPIRV/vector-to-spirv.mlir:255]: Added

**Review by kuhar — APPROVED** [review]
Thanks!

File: mlir/lib/Conversion/VectorToSPIRV/VectorToSPIRV.cpp [hunk 3]
```diff
@@ -169,9 +154,17 @@ struct VectorExtractOpConvert final
       return success();
     }
 
-    int32_t id = getFirstIntValue(extractOp.getMixedPosition());
-    rewriter.replaceOpWithNewOp<spirv::CompositeExtractOp>(
-        extractOp, adaptor.getVector(), id);
+    std::optional<int64_t> id =
+        getConstantIntValue(extractOp.getMixedPosition()[0]);
+
+    if (id.has_value())
```
Comment (kuhar) [inline mlir/lib/Conversion/VectorToSPIRV/VectorToSPIRV.cpp:160]:
```suggestion
    if (std::optional<int64_t> id =
        getConstantIntValue(extractOp.getMixedPosition()[0]);)
```
you could do this

File: mlir/lib/Conversion/VectorToSPIRV/VectorToSPIRV.cpp [hunk 4]
```diff
@@ -249,9 +242,16 @@ struct VectorInsertOpConvert final
       return success();
     }
 
-    int32_t id = getFirstIntValue(insertOp.getMixedPosition());
-    rewriter.replaceOpWithNewOp<spirv::CompositeInsertOp>(
-        insertOp, adaptor.getSource(), adaptor.getDest(), id);
+    std::optional<int64_t> id =
+        getConstantIntValue(insertOp.getMixedPosition()[0]);
+
+    if (id.has_value())
```
Comment (kuhar) [inline mlir/lib/Conversion/VectorToSPIRV/VectorToSPIRV.cpp:248]: same here

**Pull request merged**